docs/mp/oci/01_OciObjectStorage.adoc

```diff
@@ -0,0 +1,205 @@
+    }
+----
+
+1. Use `objectStorage.putObject` method with`PutObject.Request.builder()` to submit data
+3. Put the result in the `Response`
+
+
+=== Download file
+To download the file via `GET` method from OCI Object Storage:
+
+[source,java]
+----
+    @GET
+    @Path("/file/{file-name}")
+    public Response download(@PathParam("file-name") String fileName) {
+        ApiOptionalResponse<GetObject.Response> ociResponse = objectStorage.getObject(GetObject.Request.builder() (1)
+                                                                                                      .bucket(bucketName)
+                                                                                                      .objectName(fileName));
+        Optional<GetObject.Response> entity = ociResponse.entity(); (2)
+
+        if (entity.isEmpty()) {
+            return Response.status(Response.Status.NOT_FOUND).build(); (3)
+        }
+
+        GetObject.Response response = entity.get();
+
+        StreamingOutput stream = output -> response.writeTo(Channels.newChannel(output));
+
+        Response.ResponseBuilder ok = Response.ok(stream, MediaType.APPLICATION_OCTET_STREAM_TYPE) (4)
+                .header(Http.Header.CONTENT_DISPOSITION, "attachment; filename=\"" + fileName + "\"")
+                .header("opc-request-id", ociResponse.headers().first("opc-request-id").orElse(""))
+                .header("request-id", ociResponse.requestId());
+
+        ociResponse.headers()
+                .first(Http.Header.CONTENT_TYPE)
+                .ifPresent(ok::type);
+
+        ociResponse.headers()
+                .first(Http.Header.CONTENT_LENGTH)
+                .ifPresent(it -> ok.header(Http.Header.CONTENT_LENGTH, it));
+
+        return ok.build();
+    }
+----
+1. Use `getObject` function to make asynchronous request to OCI Object Storage
+2. The result is of type `Optional`
+3. Whenever the result is empty, return status `404`
+4. Get the response, set headers and return the result
+
+
+=== Delete
+Finally, to delete a file, `DELETE` method should be used
+
+[source,java]
+----
+    @DELETE
+    @Path("/file/{file-name}")
+    public Response delete(@PathParam("file-name") String fileName) {
+        DeleteObject.Response response = objectStorage.deleteObject(DeleteObject.Request.builder() (1)
+                                                                            .bucket(bucketName)
+                                                                            .objectName(fileName));
+
+        return Response.status(response.status().code()) (2)
+                .header("opc-request-id", response.headers().first("opc-request-id").orElse(""))
+                .header("request-id", response.requestId())
+                .build();
+    }
+----
+
+1. Use `deleteObject` function and configure a `DeleteObject.Request.builder()` to submit the delete request
+2. Return the result
+
+== Using CURL and WGET to work with files stored in OCI
```

Comment:
I am not sure this should be defined in documentation, this seems more a guide thing. 
I think we should have a list/table of available methods on the API (possibly with links to OCI documentation of that API call and to Javadoc of the request/response classes, if we can construct the links right now).
This is for any of the documents - documentation should focus more on what we provide than on examples. So there should be a simple example of injection + a list of what can be injected, rather than full blown upload/download example.

Reply:
removed